st2api/tests/unit/controllers/v1/test_actions.py

```diff
@@ -370,6 +370,36 @@ def test_get_all_exclude_attributes(self):
         self.__do_delete(action_1_id)
         self.__do_delete(action_2_id)
 
+    @mock.patch.object(action_validator, 'validate_action', mock.MagicMock(
+        return_value=True))
+    def test_get_all_include_attributes(self):
+        action_1_id = self.__get_action_id(self.__do_post(ACTION_1))
+        action_2_id = self.__get_action_id(self.__do_post(ACTION_2))
+
+        # Invalid include attribute
+        resp = self.app.get('/v1/actions?include_attributes=invalid',
+                            expect_errors=True)
+        self.assertEqual(resp.status_int, 400)
+        self.assertTrue('Invalid or unsupported include attribute specified' in
+                        resp.json['faultstring'])
+
+        # Valid include attribute
+        resp = self.app.get('/v1/actions?include_attributes=name')
+        self.assertEqual(resp.status_int, 200)
+        self.assertEqual(len(resp.json), 2, '/v1/actions did not return all actions.')
+        self.assertFalse(resp.json[0]['entry_point'])
+        self.assertFalse(resp.json[1]['entry_point'])
+
+        # Valid include attribute
+        resp = self.app.get('/v1/actions?include_attributes=entry_point')
+        self.assertEqual(resp.status_int, 200)
+        self.assertEqual(len(resp.json), 2, '/v1/actions did not return all actions.')
+        self.assertTrue(resp.json[0]['entry_point'])
+        self.assertTrue(resp.json[1]['entry_point'])
+
+        self.__do_delete(action_1_id)
+        self.__do_delete(action_2_id)
+
```

Comment:
Unit test for the exclude and include attributes mutual exclusion error and unit test for malformed include_attributes in the API call?

Reply:
Good catch - forgot those, will add it.

Reply:
Just had a look - unit test for malformed `include_attributes` is already there (see first test case / scenario in this method), added one for mutual exclusivity in 5d8d3d74a4e63e951f0e273e367716c98a591fbd, 1951ff13045a8b397c4fa782cd21fc674a9c310a.